lib/remote/apilistener.cpp

```diff
@@ -1532,19 +1534,31 @@ void ApiListener::ReplayLog(const JsonRpcConnection::Ptr& client)
 						continue;
 				}
 
+				bool sent;
+
 				try  {
-					client->SendRawMessage(pmessage->Get("message"));
-					count++;
+					sent = client->SendRawMessage(pmessage->Get("message"));
 				} catch (const std::exception& ex) {
 					Log(LogWarning, "ApiListener")
 						<< "Error while replaying log for endpoint '" << endpoint->GetName() << "': " << DiagnosticInformation(ex, false);
 
 					Log(LogDebug, "ApiListener")
 						<< "Error while replaying log for endpoint '" << endpoint->GetName() << "': " << DiagnosticInformation(ex);
 
+					stopReplay = true;
+					break;
+				}
+
+				if (!sent) {
+					Log(LogInformation, "ApiListener")
+						<< "Endpoint '" << endpoint->GetName() << "' disconnected while replaying log.";
+
+					stopReplay = true;
 					break;
 				}
 
+				++count;
+
```

Comment:
It just looks a bit awkward and the name "sent" is not very informative either. I therefore suggest changing this to something like this, which is very compact and doesn't need intermediate flags.

```cxx
				try  {
					if (!client->SendRawMessage(pmessage->Get("message"))) {
						Log(LogInformation, "ApiListener")
							<< "Endpoint '" << endpoint->GetName() << "' disconnected while replaying log.";

						stopReplay = true;
						break;
					}

					++count;
				} catch (const std::exception& ex) {
					...
				}
``